src/main/java/com/cloudbees/jenkins/support/impl/DumpExportTable.java

```diff
@@ -24,53 +25,94 @@
  * memory leaks.
  *
  * @since 2.23
- *
+ * <p>
  * User: schristou88
  * Date: 1/8/15
  * Time: 3:54 PM
  */
 @Extension
-public class DumpExportTable extends Component {
-  private final Logger logger = Logger.getLogger(DumpExportTable.class.getName());
+public class DumpExportTable extends ObjectComponent<Computer> {
+
+    private static final long serialVersionUID = 1L;
+    
+    private final Logger logger = Logger.getLogger(DumpExportTable.class.getName());
+
+    @DataBoundConstructor
+    public DumpExportTable() {
+    }
+
+    @NonNull
+    @Override
+    public Set<Permission> getRequiredPermissions() {
+        return Collections.singleton(Jenkins.ADMINISTER);
+    }
+
+    @NonNull
+    @Override
+    public String getDisplayName() {
+        return "Dump agent export tables (could reveal some memory leaks)";
+    }
 
-  @NonNull
-  @Override
-  public Set<Permission> getRequiredPermissions() {
-    return Collections.singleton(Jenkins.ADMINISTER);
-  }
+    @Override
+    public void addContents(@NonNull Container result) {
+        Jenkins.get().getNodes().stream()
+                .map(Node::toComputer)
+                .forEach(computer -> Optional.ofNullable(computer)
+                        .ifPresent(comp -> addContents(result, comp))
+                );
+    }
+
+    @Override
+    public void addContents(@NonNull Container container, @NonNull Computer item) {
+        container.add(
+                new TruncatedContent("nodes/slave/{0}/exportTable.txt", item.getName()) {
+                    @Override
+                    protected void printTo(PrintWriter out) {
+                        try {
+
+                            VirtualChannel channel = item.getChannel();
+                            if (channel instanceof Channel) // Should never be false but just in case.
+                            {
+                                ((Channel) channel).dumpExportTable(out);
+                            }
+                        } catch (TruncationException e) {
+                            logger.log(Level.WARNING,
+                                    "Truncated the output of export table for node: " + item.getName(), e);
+                        } catch (IOException e) {
+                            logger.log(Level.WARNING,
+                                    "Could not record environment of node " + item.getName(), e);
+                        }
+                    }
+                }
+        );
+    }
 
-  @NonNull
-  @Override
-  public String getDisplayName() {
-    return "Dump agent export tables (could reveal some memory leaks)";
-  }
+//  @Override
```

Comment:
Why is this commented?

Reply:
Should be assignable to Jenkins and Computer.